src/jit/morph.cpp

```diff
@@ -6950,34 +7062,75 @@ bool Compiler::fgCanFastTailCall(GenTreeCall* callee)
     }
 #endif
 
+    auto reportFastTailCallDecision = [this, callee](const char* msg, size_t callerStackSize, size_t calleeStackSize) {
+#if DEBUG
+        if ((JitConfig.JitReportFastTailCallDecisions()) == 1)
+        {
+            if (callee->gtCallType != CT_INDIRECT)
+            {
+                const char* methodName;
+
+                methodName = eeGetMethodFullName(callee->gtCallMethHnd);
+
+                printf("[Fast tailcall decision]: Caller: %s\n[Fast tailcall decision]: Callee: %s -- Decision: ",
+                       info.compFullName, methodName);
+            }
+            else
+            {
+                printf("[Fast tailcall decision]: Caller: %s\n[Fast tailcall decision]: Callee: IndirectCall -- "
+                       "Decision: ",
+                       info.compFullName);
+            }
+
+            if (callerStackSize != -1)
+            {
+                printf("[Fast tailcall decision]: %s (CallerStackSize: %d, CalleeStackSize: %d)\n\n", msg,
+                       callerStackSize, calleeStackSize);
+            }
+            else
+            {
+                printf("[Fast tailcall decision]: %s\n\n", msg);
+            }
```

Comment:
These last two shouldn't print "[Fast tailcall decision]: " since they are printed on the same line as one of the two printfs that precede them.

Also, you should add this tag to the JITDUMP below, as well